### ISSUE TITLE
build failure with gcc 11

### ISSUE THREAD

**Issue by emollier** [andersen-lab/ivar#108]
Greetings,

_ivar_ fails to build from source when built with Gcc 11.  One can reproduce the behavior using the following steps:
1. install gcc-11 (from Debian Experimental currently),
2. export CC=gcc-11 and CXX=g++-11,
3. trigger a build of _ivar_,
4. and see the error below:

```
GNU C++11 (Debian 11-20210227-1) version 11.0.1 20210227 (experimental) [master revision ded7355b3d9:aa8bb3da10e:06a9f20f97e788bdff139dcf62d055955debcc8d] (x86_64-linux-gnu)
	compiled by GNU C version 11.0.1 20210227 (experimental) [master revision ded7355b3d9:aa8bb3da10e:06a9f20f97e788bdff139dcf62d055955debcc8d], GMP version 6.2.1, MPFR version 4.1.0, MPC version 1.2.0, isl version isl-0.23-GMP

GGC heuristics: --param ggc-min-expand=30 --param ggc-min-heapsize=4096
Compiler executable checksum: 2900c290e29b54415706f52c8eebe364
ivar.cpp: In function â€˜int main(int, char**)â€™:
ivar.cpp:481:11: error: â€˜void free(void*)â€™ called on pointer returned from a mismatched allocation function [-Werror=mismatched-new-delete]
  481 |       free(files);
      |       ~~~~^~~~~~~
ivar.cpp:461:33: note: returned from â€˜void* operator new [](std::size_t)â€™
  461 |     char **files = new char*[100];
      |                                 ^
```

The operating system in use is:
  - OS: Debian,
  - Version Experimental.

This has been first caught by Matthias Klose in [Debian bug #984067](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=984067), but the report misses unfortunately the interesting error message in the build log extract.  I wrapped up a quick patch/workaround to resolve the issue, available on [Debian Salsa](https://salsa.debian.org/med-team/ivar/-/blob/11b193efcf1a4a5402c13351d3bfbf457cc1b246/debian/patches/gcc-11.patch), feel free to use it if you think it is fit.

There is no emergency yet, since _gcc-11_ is not yet part of the development branch of Debian; this is mostly for information.

Have a nice day,  :)
Étienne.

**Comment by gkarthik** [comment]
Hey @emollier , thank you for pointing this out! On my radar.

**Comment by LocutusOfBorg** [comment]
Hello, we have now another failure, please have a look at the #984067 debian bug, I posted a patch there.
failure log:
https://launchpad.net/ubuntu/+source/ivar/1.3.1+dfsg-2

fixed build:
https://launchpad.net/ubuntu/+source/ivar/1.3.1+dfsg-2ubuntu1

patch:
http://launchpadlibrarian.net/559658027/ivar_1.3.1+dfsg-2_1.3.1+dfsg-2ubuntu1.diff.gz

**Comment by LocutusOfBorg** [comment]
However there are test failures on armhf, ppc64el, s390x

**Comment by cmaceves** [comment]
addressed in PR #136
